File: CultureLinkCRM/CultureLinkCRM.Web/Modules/ServerTypes/CrmDB/ZipCodeColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ZipCodeRow } from "./ZipCodeRow";

export interface ZipCodeColumns {
    ZipCodeId: Column<ZipCodeRow>;
    ZipCode: Column<ZipCodeRow>;
    ListInd: Column<ZipCodeRow>;
}

export class ZipCodeColumns extends ColumnsBase<ZipCodeRow> {
    static readonly columnsKey = 'CrmDB.ZipCode';
    static readonly Fields = fieldsProxy<ZipCodeColumns>();
}

File: CultureLinkCRM/CultureLinkCRM.Web/Modules/CrmDB/ZipCode/ZipCodeColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace CultureLinkCRM.CrmDB.Columns;

[ColumnsScript("CrmDB.ZipCode")]
[BasedOnRow(typeof(ZipCodeRow), CheckNames = true)]
public class ZipCodeColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int ZipCodeId { get; set; }
    [EditLink]
    public string ZipCode { get; set; }
    public int ListInd { get; set; }
}